src/pages/Admin/Venue/Rooms/RoomsForm.tsx

```diff
@@ -135,11 +135,13 @@ const RoomInnerForm: React.FC<RoomInnerForm> = (props) => {
 
   const onSubmit = useCallback(
     async (vals: Partial<FormValues>) => {
-      if (!user || !editingRoom) return;
+      if (!user) return;
+
       try {
         const roomValues: RoomInput = {
           ...editingRoom,
-          ...vals,
+          // @debt remove as, when properly implement room values
+          ...(vals as RoomInput),
```

Comment:
Why was the `as RoomInput` needed here?

Reply:
@yurii-lubynets could you please look into this & fix it properly so that we don't need to use `as`?

@0xdevalias apparently, `vals` doesn't match the `RoomInput` type(lacks some properties if I'm not mistaken), but I don't know the exact reason(and didn't have time to dive deeply into the logic of that) and thus added that `@debt` comment.

Reply:
@mike-lvov @0xdevalias yeah, I'm doing PR today and will re-check the issue

Reply:
here are the improvements for create/edit room: https://github.com/sparkletown/sparkle/pull/1338